meteor/client/lib/forms/schemaFormWithOverrides.tsx

```diff
@@ -0,0 +1,281 @@
+import { JsonTextInputControl } from '../Components/JsonTextInput'
+import {
+	LabelAndOverrides,
+	LabelAndOverridesForCheckbox,
+	LabelAndOverridesForDropdown,
+	LabelAndOverridesForInt,
+} from '../Components/LabelAndOverrides'
+import { MultiLineTextInputControl } from '../Components/MultiLineTextInput'
+import { TextInputControl } from '../Components/TextInput'
+import { JSONSchema, TypeName } from '@sofie-automation/shared-lib/dist/lib/JSONSchemaTypes'
+import { SchemaFormTable } from './schemaFormTable'
+import { joinObjectPathFragments, translateStringIfHasNamespaces } from './schemaFormUtil'
+
+export interface SchemaFormCommonProps {
+	schema: JSONSchema
+	translationNamespaces: string[]
+
+	/**
+	 * In some situations, it is desirable to not allow tables to be used by a schema
+	 * For example, a table inside a table will not display properly so this gets set automatically
+	 */
+	allowTables?: boolean
+}
+
+interface SchemaFormWithOverridesProps extends SchemaFormCommonProps {
+	attr: string
+
+	item: WrappedOverridableItemNormal<any>
+	overrideHelper: OverrideOpHelper
+}
+
+interface FormComponentProps {
+	schema: JSONSchema
+	translationNamespaces: string[]
+	commonAttrs: {
+		label: string
+		hint: string | undefined
+
+		item: WrappedOverridableItemNormal<any>
+		overrideHelper: OverrideOpHelper
+		itemKey: string
+		opPrefix: string
+	}
+}
+
+function useChildPropsForFormComponent(props: SchemaFormWithOverridesProps) {
+	return useMemo(() => {
+		const title = props.schema['ui:title'] || props.attr
+		const description = props.schema['ui:description']
+
+		return {
+			schema: props.schema,
+			translationNamespaces: props.translationNamespaces,
+			commonAttrs: {
+				label: translateStringIfHasNamespaces(title, props.translationNamespaces),
+				hint: description ? translateStringIfHasNamespaces(description, props.translationNamespaces) : undefined,
+				item: props.item,
+				itemKey: props.attr,
+				opPrefix: props.item.id,
+				overrideHelper: props.overrideHelper,
+			},
+		}
+	}, [props.schema, props.translationNamespaces, props.attr, props.item, props.overrideHelper])
+}
+
+export function SchemaFormWithOverrides(props: SchemaFormWithOverridesProps) {
+	const { t } = useTranslation()
+
+	const childProps = useChildPropsForFormComponent(props)
+
+	switch (props.schema.type) {
+		case TypeName.Array:
+			return <ArrayFormWithOverrides {...props} />
+		case TypeName.Object:
+			if (props.schema['ui:displayType'] === 'json') {
+				return <JsonFormWithOverrides {...childProps} />
+			} else {
+				return <ObjectFormWithOverrides {...props} />
+			}
+		case TypeName.Integer:
+			if (props.schema.enum) {
+				return <EnumFormWithOverrides {...childProps} />
+			} else {
+				return <IntegerFormWithOverrides {...childProps} />
+			}
+		case TypeName.Number:
+			return <NumberFormWithOverrides {...childProps} />
+		case TypeName.Boolean:
+			return <BooleanFormWithOverrides {...childProps} />
+		case TypeName.String:
```

Comment:
The original Config Manifest system supported a multi-line string. I think this too should support that - maybe through a `"ui:multiline": Boolean` key, for example?

Reply:
The multiline support we had was producing an array of strings and is now handled at https://github.com/nrkno/sofie-core/blob/feat/gateways-json-schema/meteor/client/lib/forms/schemaFormWithOverrides.tsx#L115-L116

Example use: https://github.com/nrkno/sofie-timeline-state-resolver/blob/feat/config-schema/packages/timeline-state-resolver/src/integrations/vizMSE/%24schemas/options.json#L71-L77

A `ui:multiline` property would make sense, but as it would be new functionality should come later?

Reply:
Ah, okay, I thought that was just Strings with newlines. If that's the case, then yeah, I think we are just aiming at feature parity.